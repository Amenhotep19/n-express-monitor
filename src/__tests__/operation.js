import * as nAutoMetrics from '@financial-times/n-auto-metrics';

import setupMonitor from '../setup';
import monitor from '../operation';

jest.mock('@financial-times/n-auto-metrics', () => ({
	metricsOperation: jest.fn(inputFunction => inputFunction),
	initAutoMetrics: jest.fn(),
}));

describe('monitor', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	const app = {
		use: jest.fn(),
	};

	const metrics = {};

	it('disable metricsOperation if metrics instance is not set', () => {
		setupMonitor({ app });
		const targetOperation = () => {};
		monitor(targetOperation);
		expect(nAutoMetrics.metricsOperation.mock.calls).toMatchSnapshot();
	});

	it('enable metricsOperation if metrics instance is set', () => {
		setupMonitor({ app, metrics });
		const targetOperation = () => {};
		monitor(targetOperation);
		expect(nAutoMetrics.metricsOperation.mock.calls).toMatchSnapshot();
	});
});
