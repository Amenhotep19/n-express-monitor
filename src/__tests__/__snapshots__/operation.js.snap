// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monitor disable metricsOperation if metrics instance is not set 1`] = `Array []`;

exports[`monitor enable metricsOperation if metrics instance is set 1`] = `
Array [
  Array [
    [Function],
  ],
]
`;
